src/test/javascript/spec/component/overview/exercise-details/course-exercise-details.component.spec.ts

```diff
@@ -250,4 +275,34 @@ describe('CourseExerciseDetailsComponent', () => {
         expect(comp.wasSubmissionSimulated).toBe(false);
         expect(comp.exercise?.participationStatus).toBe(ParticipationStatus.EXERCISE_SUBMITTED);
     }));
+
+    it('should fill & empty sample modeling solution', () => {
+        comp.showIfExampleSolutionPresent({ ...modelingExercise });
+        jestExpect(comp.exampleSolution).toBe(undefined);
+        jestExpect(comp.exampleSolutionUML).not.toBe(undefined);
```

Comment:
Can you expect an actual attribute of the `modelingExercise` here instead of ‘not undefined’?

Same for the tests below.

Reply:
I made these 3 tests more specific, can you take a look again?

Reply:
Thanks, for the modelling exercise it looks good now.

For the other two: Where does the `.changingThisBreaksApplicationSecurity` come from? There are a lot of annotations by Github Actions that using it is not a good idea. Additionally, the client tests fail due to this.

Reply:
`.changingThisBreaksApplicationSecurity` is a result of calling `ArtemisMarkdownService.safeHtmlForMarkdown` (that internally calls `DomSanitizer.bypassSecurityTrustHtml`) which is set on `SafeHtml` objects but it seems like this is not a public property. Since `SafeHtml` is an Angular interface and not a custom one and it doesn't expose any public properties, maybe I can convert Text and File Upload exercise tests to their old version: `expect(comp.exampleSolution).not.toBe(undefined)`. What do you think?

Reply:
Sure, if there is no other useful assertion that can be made here, then using `not.toBe(undefined)` is okay for those tests.

Reply:
Thanks, I committed these changes.